compiler/src/iree/compiler/Codegen/Common/GPU/GPUInferMemorySpace.cpp

```diff
@@ -38,6 +39,12 @@ bool isDefinitelyShared(bufferization::AllocTensorOp alloc) {
   // thread distributed `scf.forall` op. All other shared allocations are
   // expected to be properly indicated in advance.
   for (auto user : alloc->getUsers()) {
+    if (auto linalgCopy = dyn_cast<linalg::CopyOp>(user)) {
+      if (auto useDMAConfig =
+              getLoweringConfig<IREE::GPU::UseGlobalLoadDMAAttr>(linalgCopy))
+        return true;
```

Comment:
This needs to hold for all users, so continue the loop instead.
```suggestion
        continue;
```